peer/x/roundrobin/list.go

```diff
@@ -55,27 +55,28 @@ type List struct {
 	started            atomic.Bool
 }
 
-func (pl *List) addAll(peerIDs []peer.Identifier) error {
+// Update applies the additions and removals of peer Identifiers to the list
+// it returns a multi-error result of every failure that happened without
+// circuit breaking due to failures
+func (pl *List) Update(additions, removals []peer.Identifier) error {
 	pl.lock.Lock()
 	defer pl.lock.Unlock()
```

Comment:
Will this ever be called with both lists being empty? Is it worth returning
early in that case (before even acquiring the lock)?